### PR TITLE
Clean up build.py

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 
1. Delete the data downloading functions (they are not used in CI build anymore). If you still need them, please use the [download_test_data.py](https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/download_test_data.py).
2. Enable model test for each provider
3. Disable Arena allocator for x86_32 build because it may run into infinite loop when integer overflow happens. 
4. Make onnx_test_runner.exe's  error message more verbose. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -260,53 +257,9 @@ def check_md5(filename, expected_md5):
         return False
     return True
 
-#the last part of src_url should be unique, across all the builds
-def download_test_data(build_dir, src_url, expected_md5, azure_sas_key):
```
Comment (skottmckay) [inline tools/ci_build/build.py]: Is this not useful to a developer who needs to get the test data?

Comment (snnn) [reply tools/ci_build/build.py]: The developer need more than 50gb disk space for doing that, so far I didn't see anyone that want to try it locally.

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -995,29 +945,34 @@ def main():
...
               run_onnx_tests(build_dir, configs, onnx_test_data_dir, 'nuphar', args.enable_multi_device_test, False, 1)
-            else:
-              run_onnx_tests(build_dir, configs, onnx_test_data_dir, None, args.enable_multi_device_test, True, 0)
 
             if args.use_dml:
-              run_onnx_tests(build_dir, configs, onnx_test_data_dir, 'dml', args.enable_multi_device_test, False, 1)
+              run_onnx_tests(build_dir, configs, onnx_test_data_dir, 'dml', args.enable_multi_device_test, False, 1)  
+
+			#OOM because of memory leak
+            #if args.use_mkldnn:
```
Comment (skottmckay) [inline tools/ci_build/build.py]: nit: tabs

File: tools/ci_build/build.py (hunk 1 above)
Comment (skottmckay) [reply tools/ci_build/build.py]: I have them locally but can't remember if I used this or tools\ci_build\github\download_test_data.py

I found the latter easily enough just now so I'm fine with this being removed from build.py

---
In reply to: [348787462](https://github.com/microsoft/onnxruntime/pull/2446#discussion_r348787462) [](ancestors = 348787462)

**Pull request merged**